File: src/_02_StructuralPattern/_02_05_Facade/java/after/EmailMessageFacade.java
package _02_StructuralPattern._02_05_Facade.java.after;

public class EmailMessageFacade {
  
  private String from;
  private String to;
  private String subject;
  private String text;

  // from
  public String getFrom() {
    return from;
  }
  public void setFrom(String from) {
    this.from = from;
  }

  // to
  public String getTo() {
    return to;
  }
  public void setTo(String to) {
    this.to = to;
  }

  // subject
  public String getSubject() {
    return subject;
  }
  public void setSubject(String subject) {
    this.subject = subject;
  }

  // text
  public String getText() {
    return text;
  }
  public void setText(String text) {
    this.text = text;
  }
  
}

File: src/_02_StructuralPattern/_02_05_Facade/EmailMessageFacade.js
export default class EmailMessageFacade {
  /** @type { string } */
  #from;

  /** @type { string } */
  #to

  /** @type { string } */
  #subject;

  /** @type { string } */
  #text;

  // from
  /** @returns { string } */
  getFrom() {
    return this.#from;
  }
  /** @param { string } from */
  setFrom(from) {
    this.#from = from;
  }

  // to
  /** @returns { string } */
  getTo() {
    return this.#to;
  }
  /** @param { string } to */
  setTo(to) {
    this.#to = to;
  }

  // subject
  /** @returns { string } */
  getSubject() {
    return this.#subject;
  }
  /** @param { string } */
  setSubject(subject) {
    this.#subject = subject;
  }

  // text
  /** @returns { string } */
  getText() {
    return this.#text;
  }
  /** @param { string } text */
  setText(text) {
    this.#text = text;
  }
}